lib/blpapi-wrapper.ts

```diff
@@ -99,7 +116,8 @@ function subscriptionsToServices(subscriptions: Subscription[]): string[] {
         // XXX: note that we shoud probably capture what the default service is to use when
         //      reading in the session options.  However, when not specified, it is
         //      '//blp/mktdata'.
-        return match ? match[0] : '//blp/mktdata';
+        subscription.serviceUri = match ? match[0] : '//blp/mktdata';
```

Comment:
Yeah, it looks like it is.  Let's see if we can find a different approach that doesn't modify the passed-in subscriptions from the caller.